### PR TITLE
Clean up the algorithm base classes and run() methods.

### PR DESCRIPTION
**Pull request by ttung**
1. Ensure that the algorithm base classes's `run()` method is consistent with the implementations.
2. Fix missing documentation.
3. Add `*args` to all the run methods such that the implementations' run methods can take additional parameters.

Test plan: `make -j fast lint mypy`

### CODE REVIEW

File: starfish/image/_filter/clip.py [hunk 1]
```diff
@@ -59,8 +59,12 @@ def _clip(image: Union[xr.DataArray, np.ndarray], p_min: int, p_max: int) -> np.
...
 
     def run(
-            self, stack: ImageStack, in_place: bool=False, verbose: bool=False,
-            n_processes: Optional[int]=None
+            self,
+            stack: ImageStack,
+            in_place: bool=False,
+            verbose: bool=False,
+            n_processes: Optional[int]=None,
+            *args,
```
Comment (shanaxel42) [inline starfish/image/_filter/clip.py]: where would we actually use these extra *args? Should we pass them into image.clip as well?

Comment (shanaxel42) [inline starfish/image/_filter/clip.py]: where would we actually use these extra *args? Should we pass them into image.clip as well?

File: starfish/image/_filter/linear_unmixing.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 class LinearUnmixing(FilterAlgorithmBase):
 
-    def __init__(self, coeff_mat: np.ndarray, clip_method: int=1) -> None:
+    def __init__(self, coeff_mat: np.ndarray) -> None:
```
Comment (shanaxel42) [inline starfish/image/_filter/linear_unmixing.py]: was clip_method supposed to be deleted?

File: starfish/image/_filter/clip.py (hunk 1 above)
Comment (ttung) [reply starfish/image/_filter/clip.py]: This is to make the signature match the superclass.

File: starfish/image/_filter/linear_unmixing.py (hunk 2 above)
Comment (ttung) [reply starfish/image/_filter/linear_unmixing.py]: It's unused.

Comment (ttung) [reply starfish/image/_filter/linear_unmixing.py]: @kevinyamauchi I think you added this.  Let me know if it's a problem.

**Pull request merged**